earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/shared/filewalker_unittest.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Unit Tests for walking globe files.
+
+#include <algorithm>
+#include <limits.h>
+#include <map>
+#include <string>
+#include <utility>
+#include <gtest/gtest.h>
+#include "common/khFileUtils.h"
+
+#include "fusion/portableglobe/servers/fileunpacker/shared/glc_unpacker.h"
+#include "fusion/portableglobe/servers/fileunpacker/shared/portable_glc_reader.h"
+
+// Test classes for file walker package.
+class FileWalkerTest : public testing::Test {
+ protected:
+  std::string glb_file;
+  std::string glc_file;
+
+  virtual void SetUp() {
+    glb_file = "../../../fusion/portableglobe/servers/fileunpacker/test_data/test.glb";
+    glc_file = "../../../fusion/portableglobe/servers/fileunpacker/test_data/test.glc";
+  }
+
+  // Helper function to find values in lists
+  template<typename T>
+  bool contains(std::list<T> & element_list, const T & element)
+  {
+	  auto it = std::find(element_list.begin(), element_list.end(), element);
+	  return it != element_list.end();
+  }
+
+  /**
+    * Test that the contents of the test GLB have expected size and values
+    */
+  void TestGlbContents() {
+    PortableGlcReader reader(glb_file.c_str());
+    GlcUnpacker unpacker(reader, false, false);
+    std::list<std::string> files;
+    unpacker.MapFileWalker(0, [&] (int layer, const std::string& file_name) {
```

Comment:
The second parameter should be `const char *` .  I'm surprised this even compiled but it did.  I think it must be a bug in the compiler because it should of failed.